src/components/expanded-state/useSwapDetailsButtonPosition.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+} from 'react-native-reanimated';
+import { useCallback, useMemo, useRef, useState } from 'react';
+
+export default ({ contentHeight }: { contentHeight: number }) => {
+  const offsetButtonAnimation = useSharedValue(0);
+  const justOpenedSheet = useRef(false);
+  const [buttonPosition, setButtonPosition] = useState<number | undefined>();
+
+  const onHeightChange = useCallback(
+    event => {
+      if (justOpenedSheet.current) {
+        offsetButtonAnimation.value = withSpring(
+          event?.nativeEvent?.layout?.height - contentHeight,
+          {
+            damping: 10,
+            mass: 0.08,
+            stiffness: 100,
+            overshootClamping: true,
+          }
+        );
+      }
+    },
+    [contentHeight, offsetButtonAnimation]
+  );
+
+  const buttonWrapperStyle = useAnimatedStyle(() => ({
+    transform: [{ translateY: offsetButtonAnimation.value }],
+  }));
+
+  const wrapperStyle = useMemo(
+    () =>
+      ios && [
+        buttonWrapperStyle,
+        buttonPosition && {
+          position: 'absolute',
+          top: buttonPosition,
+          width: '100%',
+        },
+      ],
+    [buttonPosition, buttonWrapperStyle]
+  );
+
+  const onPressMore = useCallback(() => {
+    justOpenedSheet.current = true;
+  }, []);
+
+  const onWrapperLayout = useCallback(
+    e => !justOpenedSheet.current && setButtonPosition(e.nativeEvent.layout.y),
+
+    []
+  );
+
+  return {
+    wrapperStyle,
+    buttonWrapperStyle,
+    onHeightChange,
+    justOpenedSheet,
+    onPressMore,
+    onWrapperLayout,
```

Comment:
actually `buttonWrapperStyle` and `justOpenedSheet` is not used where the hook is called, don't have string opinions about removing if we're not using so not blocker

Reply:
ah right, thanks!